test/org/sagebionetworks/bridge/cache/CacheProviderMockTest.java

```diff
@@ -171,39 +162,52 @@ public void testGetUserSessionByUserId() throws Exception {
         CacheProvider mockCacheProvider = spy(cacheProvider);
         mockCacheProvider.getUserSessionByUserId(USER_ID);
         
-        verify(jedisOps).get("userId:session:user");
-        verify(jedisOps).get("sessionToken:session");
+        verify(jedisOps).get("userId:session2:user");
     }
 
     @Test
     public void testRemoveSession() {
         StudyParticipant participant = new StudyParticipant.Builder().withId(USER_ID).build();
 
         UserSession session = new UserSession(participant);
-        session.setSessionToken(SESSION_TOKEN);
+        session.setSessionToken(DECRYPTED_SESSION_TOKEN);
         
         cacheProvider.removeSession(session);
-        cacheProvider.getUserSession(SESSION_TOKEN);
+        cacheProvider.getUserSession(DECRYPTED_SESSION_TOKEN);
         
-        verify(transaction).del(SESSION_KEY.toString());
-        verify(transaction).del(USER_SESSION_KEY.toString());
+        verify(transaction).del(TOKEN_TO_USER_ID.toString());
+        verify(transaction).del(USER_ID_TO_SESSION.toString());
         verify(transaction).exec();
     }
 
     @Test
-    public void testRemoveSessionByUserId() {
+    public void testRemoveSessionByUserId() throws Exception {
+        UserSession session = createUserSession();
+        String ser = BridgeObjectMapper.get().writeValueAsString(session);
+
+        when(jedisOps.get(USER_ID_TO_SESSION.toString())).thenReturn(ser);
+        
         cacheProvider.removeSessionByUserId(USER_ID);
         
-        verify(transaction).del(SESSION_KEY.toString());
-        verify(transaction).del(USER_SESSION_KEY.toString());
+        verify(transaction).del(TOKEN_TO_USER_ID.toString());
+        verify(transaction).del(USER_ID_TO_SESSION.toString());
```

Comment:
Similarly, this should verify we delete all 4 keys.

Reply:
Added